### PR TITLE
[BUG] Cannot remove unused file

### PR DESCRIPTION
**Pull request by balanza**
This PR demonstrates an alleged bug in the project toolchain. We are moving and renaming files around the `apps/io-services-cms-webapp` project to have more clarity. 

The issue arises as soon we move or delete one specific file (`src/apim_client.ts`). Please note that the file can be renamed and duplicated as long as it holds its original location.

### Changes

We made changes in two commits. The first commit builds smoothly, the second breaks the build.

In the [first commit](https://github.com/pagopa/io-services-cms/commit/ff1a284393414a38f246df184d288fdebf2e81b1):
* moved and rename all files but `src/apim_client.ts`
* no references point to `src/apim_client.ts` anymore
* `src/apim_client.ts` has been renamed to `src/NOT-USED-apim-client.ts`

In the [second commit](https://github.com/pagopa/io-services-cms/commit/e3ffa40b3c809cc682ea5770f76fbdef2b95a4a7) we just deleted `src/NOT-USED-apim-client.ts`, provoking the following build error:
```sh
io-services-cms-webapp:build: cache miss, executing 57c702dccf6808d6
io-services-cms-webapp:build: src/webservice/controllers/create-service/index.ts(140,5): error TS2345: Argument of type 'HKT<"TaskEither", { [x: string]: any; }>' is not assignable to parameter of type 'TaskEither<Error, { userId: string & INonEmptyStringTag; productId: string & INonEmptyStringTag; }>'.
io-services-cms-webapp:build:   Type 'HKT<"TaskEither", { [x: string]: any; }>' provides no match for the signature '(): Promise<Either<Error, { userId: string & INonEmptyStringTag; productId: string & INonEmptyStringTag; }>>'.
io-services-cms-webapp:build: src/webservice/controllers/create-service/index.ts(141,7): error TS2322: Type 'TaskEither<Error, string & INonEmptyStringTag>' is not assignable to type 'HKT<"TaskEither", any>'.
io-services-cms-webapp:build: src/webservice/controllers/create-service/index.ts(142,7): error TS2322: Type 'TaskEither<Error, string & INonEmptyStringTag>' is not assignable to type 'HKT<"TaskEither", any>'.
io-services-cms-webapp:build: src/webservice/controllers/info.ts(16,14): error TS2742: The inferred type of 'makeInfoHandler' cannot be named without a reference to '@io-services-cms/models/node_modules/fp-ts/lib/Task'. This is likely not portable. A type annotation is necessary.
io-services-cms-webapp:build: ERROR: command finished with error: command (/Users/emanuele/Workspace/io/io-services-cms/apps/io-services-cms-webapp) yarn run build exited (1)
```

### Reproduce
Just checkout one of the commits and then 
```
yarn build
```

### CODE REVIEW

**Comment by balanza** [comment]
@grausof thanks for your commit. To have a single version of the dependency is a great hint, we still have two:
```sh
yarn why fp-ts
├─ @io-services-cms/models@workspace:packages/io-services-cms-models
│  └─ fp-ts@npm:2.15.0 (via npm:^2.13.1)
│
├─ @pagopa/io-functions-commons@npm:28.2.0
│  └─ fp-ts@npm:2.15.0 (via npm:^2.10.5)
│
├─ @pagopa/io-functions-commons@npm:28.2.0 [c85c2]
│  └─ fp-ts@npm:2.15.0 (via npm:^2.10.5)
│
├─ @pagopa/ts-commons@npm:10.15.0
│  └─ fp-ts@npm:2.13.1 (via npm:^2.11.0)
│
├─ @pagopa/ts-commons@npm:11.0.0
│  └─ fp-ts@npm:2.13.1 (via npm:^2.11.0)
│
└─ io-services-cms-webapp@workspace:apps/io-services-cms-webapp
   └─ fp-ts@npm:2.15.0 (via npm:^2.15.0)
```
I wonder if we can use only one version

**Comment by balanza** [comment]
Yesterday we came up with the workaround in #47, thus we left the unremovable file in the codebase (otherwise, the build breaks). Today I just removed the file in https://github.com/pagopa/io-services-cms/pull/48/commits/7675e8e25917c053d4b5ce4ca6c2cf60dec46b31 and everything works 🤷 

Besides having the bug solved, I wonder what's the root cause. Is there a hidden cache? Is it based on the current date (hence today I found no cache)?

**Comment by grausof** [comment]
> Yesterday we came up with the workaround in #47, thus we left the unremovable file in the codebase (otherwise, the build breaks). Today I just removed the file in [7675e8e](https://github.com/pagopa/io-services-cms/commit/7675e8e25917c053d4b5ce4ca6c2cf60dec46b31) and everything works 🤷
> 
> Besides having the bug solved, I wonder what's the root cause. Is there a hidden cache? Is it based on the current date (hence today I found no cache)?

did you delete `node_modules` yesterday?

**Comment by balanza** [comment]
> did you delete `node_modules` yesterday?

Yes. Also, the pipeline failed and that should be a fresh start, isn't it?

**Comment by grausof** [comment]
> > did you delete `node_modules` yesterday?
> 
> Yes. Also, the pipeline failed and that should be a fresh start, isn't it?

I don't understand what you mean when you say: Yesterday we came up with the workaround in #47 .... (otherwise, the build breaks). I don't see any PRs that went bust yesterday, soI'm guessing it wasn't just working locally.

**Comment by balanza** [comment]
**UPDATE** Another iteration on the understanding of this bug. 
I played with the `fp-ts` version and I tried two commits, one that works and one that does not.

> TL;DR The only difference seems to be the version of `fp-ts` **on the webapp** application

Below are the details of the actual version used for each commit. Used versions of `fp-ts` for each dependant are the same except for `io-services-cms-webapp`, which uses `2.13.1` and `2.15.0` in good and bad scenarios respectively.

Interestingly, `2.15.0` is the version used in both scenarios by `io-services-cms-models`. That is, the error occurs when those two packages use the very same versions of the same dependency and it doesn't where versions are discordant - which is counter-intuitive at least 😨.

✅ **Commit https://github.com/pagopa/io-services-cms/pull/48/commits/df8a317a5e70fd874f4ab081c4d5aae2320ab50b** 
```sh
yarn why fp-ts
├─ @io-services-cms/models@workspace:packages/io-services-cms-models
│  └─ fp-ts@npm:2.15.0 (via npm:^2.13.1)
│
├─ @pagopa/io-functions-commons@npm:28.2.0
│  └─ fp-ts@npm:2.15.0 (via npm:^2.10.5)
│
├─ @pagopa/io-functions-commons@npm:28.2.0 [c85c2]
│  └─ fp-ts@npm:2.15.0 (via npm:^2.10.5)
│
├─ @pagopa/ts-commons@npm:10.15.0
│  └─ fp-ts@npm:2.13.1 (via npm:^2.11.0)
│
├─ @pagopa/ts-commons@npm:11.0.0
│  └─ fp-ts@npm:2.13.1 (via npm:^2.11.0)
│
└─ io-services-cms-webapp@workspace:apps/io-services-cms-webapp
   └─ fp-ts@npm:2.13.1 (via npm:2.13.1)
```

🔴 **Commit https://github.com/pagopa/io-services-cms/pull/48/commits/a73d831ce0eae49c9d6ef3ed1a56fd445b026211**
```sh
yarn why fp-ts                     
├─ @io-services-cms/models@workspace:packages/io-services-cms-models
│  └─ fp-ts@npm:2.15.0 (via npm:^2.15.0)
│
├─ @pagopa/io-functions-commons@npm:28.2.0
│  └─ fp-ts@npm:2.15.0 (via npm:^2.10.5)
│
├─ @pagopa/io-functions-commons@npm:28.2.0 [c85c2]
│  └─ fp-ts@npm:2.15.0 (via npm:^2.10.5)
│
├─ @pagopa/ts-commons@npm:10.15.0
│  └─ fp-ts@npm:2.13.1 (via npm:^2.11.0)
│
├─ @pagopa/ts-commons@npm:11.0.0
│  └─ fp-ts@npm:2.13.1 (via npm:^2.11.0)
│
└─ io-services-cms-webapp@workspace:apps/io-services-cms-webapp
   └─ fp-ts@npm:2.15.0 (via npm:^2.15.0)
```
